### PR TITLE
#9628: Merge second set of binary backward op from tt_eager to TTNN

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Binary Backward Issue : #9628 [Master issue : #9322 ]

### List of ops completed
- binary_eq_bw ( without optional return support )
- binary_assign_bw
- concat_bw
- binary_le_bw
- rsub_bw
- bias_gelu_bw

**Structure**
```
ttnn
└── cpp
    └── ttnn
        └── operations
            └── ...
            └── eltwise
                └── binary
                └── binary_backward
                    ├── device
                    │   ├── binary_backward_op.cpp
                    │   ├── binary_backward_op.hpp
                    ├── binary_backward.hpp
                    └── binary_backward_pybind.hpp
```

**Test files**
- binary_eq_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py`
- binary_assign_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_assign.py`
- concat_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_concat.py`
- binary_le_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_le.py`
- rsub_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_rsub.py`
- bias_gelu_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_bias_gelu.py`

**CI :** 

- [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9712034887) - PASSED
- [[post-commit] Fast Dispatch unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9692433408) - PASSED
- [[post-commit] Slow Dispatch unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9692434538) - PASSED
- [post-commit] models tests
- Device perf regressions and output report
- Model perf regressions and output report
- Nightly fast dispatch tests

### CODE REVIEW

File: docs/source/ttnn/ttnn/ttnn/binary_le_bw.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.. _ttnn.binary_le_bw:
```
Comment (ayerofieiev-tt) [inline docs/source/ttnn/ttnn/ttnn/binary_le_bw.rst:1]: Why do we have these in tiny rst files?

File: tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py [hunk 2]
```diff
@@ -38,7 +39,8 @@ def test_bw_binary_eq(input_shapes, device):
         (torch.Size([1, 3, 320, 384])),
     ),
 )
-@pytest.mark.parametrize("are_required_outputs", [[True, True], [True, False], [False, True]])
+# @pytest.mark.parametrize("are_required_outputs", [[True, True], [True, False], [False, True]])
```
Comment (ayerofieiev-tt) [inline tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py:42]: Why do we reduce the test coverage?

Comment (VirdhatchaniKN) [reply tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py:42]: Currently we have not provided optional tensor support @ayerofieiev-tt . Hence, the output tensors cannot be null. But we have raised a separate PR for it. Once that gets merged, we can make the test file work for all cases

**Review by TT-BrianLiu — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py [hunk 3]
```diff
@@ -93,13 +96,13 @@ def test_bw_binary_eq_opt_output_qid(input_shapes, device, are_required_outputs)
 
     cq_id = 0
 
-    tt_output_tensor_on_device = tt_lib.tensor.binary_eq_bw(
+    tt_output_tensor_on_device = ttnn.binary_eq_bw(
```
Comment (eyonland) [inline tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_eq.py:99]: Again, why not clean this up now and align with pytorch naming conventions?

**Review by eyonland — APPROVED** [review]
I'm approving the PR to keep progress flowing.   Please see my comments though.  I absolutely detest how we invented new op names and did not align with the names that PyTorch has used.   We have the ability to overload these pybind11 calls.  C++ functions can also be overloaded.

File: tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_bias_gelu.py [hunk 4]
```diff
@@ -31,14 +32,11 @@ def test_bw_bias_gelu(input_shapes, approximate, device):
 
     pyt_y = torch.nn.functional.gelu(in_data, approximate=approximate)
 
-    tt_output_tensor_on_device = tt_lib.tensor.bias_gelu_bw(
-        grad_tensor, input_tensor_a, input_tensor_b, approximate=approximate
-    )
-
+    tt_output_tensor_on_device = ttnn.bias_gelu_bw(grad_tensor, input_tensor_a, input_tensor_b, approximate)
```
[the remark this answers is not in the document]
Comment (Aswinmcw) [reply tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_bias_gelu.py:35]: @eyonland , we are currently migrating the existing ops to TTNN and once it is done completely we can try overloading the ops and match the pytorch api

**Review by ayerofieiev-tt — APPROVED** [review]
Thank you!

Generally agree with Eyon.

**Pull request merged**